unicodetools/src/main/java/org/unicode/xml/UcdXML.java

```diff
@@ -0,0 +1,825 @@
+            System.exit(0);
+        }
+
+        try {
+            if (options[UCDVERSION].doesOccur) {
+                try {
+                    ucdVersion = VersionInfo.getInstance(options[UCDVERSION].value);
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not convert "
+                                    + options[UCDVERSION].value
+                                    + " to a valid UCD version");
+                }
+            } else {
+                throw new IllegalArgumentException(
+                        "Missing command line option: --ucdversion (or -v)");
+            }
+            if (options[RANGE].doesOccur) {
+                try {
+                    ucdxmloutputranges =
+                            new UCDXMLOUTPUTRANGE[] {
+                                UCDXMLOUTPUTRANGE.valueOf(
+                                        options[RANGE].value.toUpperCase(Locale.ROOT))
+                            };
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not convert "
+                                    + options[RANGE].value
+                                    + " to one of [ALL|NOUNIHAN|UNIHAN]");
+                }
+            }
+            if (options[OUTPUT].doesOccur) {
+                try {
+                    ucdxmloutputtypes =
+                            new UCDXMLOUTPUTTYPE[] {
+                                UCDXMLOUTPUTTYPE.valueOf(
+                                        options[OUTPUT].value.toUpperCase(Locale.ROOT))
+                            };
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not convert "
+                                    + options[OUTPUT].value
+                                    + " to one of [FLAT|GROUPED]");
+                }
+            }
+            if (options[OUTPUTFOLDER].doesOccur) {
+                try {
+                    destinationFolder =
+                            new File(
+                                    options[OUTPUTFOLDER].value
+                                            + getVersionString(ucdVersion, 3)
+                                            + "\\xmltest\\");
+                    if (!destinationFolder.exists()) {
+                        if (!destinationFolder.mkdir()) {
+                            throw new IOException();
+                        }
+                    }
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not find or create " + options[OUTPUTFOLDER].value);
+                }
+            } else {
+                throw new IllegalArgumentException(
+                        "Missing command line option: --outputfolder (or -f)");
```

Comment:
optional: as long as the output folder is in the "resources", it should be possible to get a File or Path to there via the class/ClassLoader APIs

Reply:
Are you expecting that the output would be in \unicodetools\data\ucdxml\ ?

This comes back to the discussion that we had in the unicodetools meeting - do we expect/require that ucdxml always is in sync with \data\ucd\ ? I think that we concluded that ucdxml will only be updated intermittently, and in which case I'd prefer to generate interim versions outside of the \unicodetools\ tree.

Reply:
That's fine.
But if we output into the "resources" path, then we should be able to get that path from Java APIs, rather than having the user tell us where it is.

Reply:
Discussed on 2/18 and implemented using Settings.Output.GEN_DIR